pycbc/waveform/waveform.py

```diff
@@ -503,6 +503,72 @@ def get_fd_waveform(template=None, **kwargs):
     params=parameters.fd_waveform_params.docstr(prefix="    ",
            include_label=False).lstrip(' '))
 
+def get_fd_waveform_from_td(**params):
+    """ Return time domain version of fourier domain approximant.
+
+    This returns a frequency domain version of a fourier domain approximant,
+    with padding and tapering at the start of the waveform.
+
+    Parameters
+    ----------
+    params: dict
+        The parameters defining the waveform to generator.
+        See `get_td_waveform`.
+
+    Returns
+    -------
+    hp: pycbc.types.FrequencySeries
+        Plus polarization time series
+    hc: pycbc.types.FrequencySeries
+        Cross polarization time series
+    """
+
+    # determine the duration to use
+    full_duration = duration = get_waveform_filter_length_in_time(**params)
+    nparams = params.copy()
+
+    while full_duration < duration * 1.5:
+        full_duration = get_waveform_filter_length_in_time(**nparams)
+        nparams['f_lower'] -= 1
+
+    if 'f_fref' not in nparams:
+        nparams['f_ref'] = params['f_lower']
+
+    # We'll try to do the right thing and figure out what the frequency
+    # end is. Otherwise, we'll just assume 2048 Hz.
+    # (consider removing as we hopefully have better estimates for more
+    # approximants
+    try:
+        f_end = get_waveform_end_frequency(**params)
+        delta_t = (0.5 / pnutils.nearest_larger_binary_number(f_end))
```

Comment:
Isn't max frequency also an optional parameter? That should take precedence if given, no?

Reply:
Yep, I'll add that in before merging.